### PR TITLE
feat: add allowances tab

### PR DESCRIPTION
**Pull request by alangsto**
## [COSMO-366](https://2u-internal.atlassian.net/browse/COSMO-366)

This PR adds an additional tab to allow instructors to add exam allowances. As of now, this is not hooked up to the backend, so the tab always renders as though there are no allowances. 

Screenshot 2024-07-16 at 2 27 59 PM

### CODE REVIEW

File: src/index.scss [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@import "@edx/brand/paragon/fonts.scss";
-@import "@edx/brand/paragon/variables.scss";
-@import "@openedx/paragon/scss/core/core.scss";
-@import "@edx/brand/paragon/overrides.scss";
+@import "~@edx/brand/paragon/fonts";
```
Comment (alangsto) [inline src/index.scss:1]: Updated this to match other MFEs.

**Review by Zacharis278 — APPROVED** [review]
LGTM 👍

File: src/pages/ExamsPage/messages.js [hunk 2]
```diff
@@ -290,6 +290,26 @@ const messages = defineMessages({
...
   },
+  allowanceDashboardTabTitle: {
+    id: 'ExamsPage.allowanceDashboardTabTitle',
+    defaultMessage: 'Allowances',
+    description: 'Title for the allowances tab',
+  },
+  allowanceButton: {
+    id: 'ExamsPage.allowanceButton',
+    defaultMessage: 'Add allowance',
+    description: 'Label for for a button to add an allowance',
```
Comment (varshamenon4) [inline src/pages/ExamsPage/messages.js:301]:
```suggestion
    description: 'Label for a button to add an allowance',
```
Nit: just noticed there's an extra for in the description.

File: src/pages/ExamsPage/index.jsx [hunk 3]
```diff
@@ -25,19 +32,29 @@ const ExamsPage = ({ courseId }) => {
   const {
     attemptsList,
   } = useExamAttemptsData();
+  const {
+    allowancesList,
+  } = useAllowancesData();
```
Comment (varshamenon4) [inline src/pages/ExamsPage/index.jsx:37]: Nice! I like how this matches with the attemptsList above.

File: src/pages/ExamsPage/index.jsx [hunk 4]
```diff
@@ -25,19 +32,29 @@ const ExamsPage = ({ courseId }) => {
...
 
   return (
     <Container>
       <Container id="exam-selector">
-        <ExamSelection exams={examsList} onSelect={setCurrentExam} />
+        <ExamSelection exams={examsList} onSelect={setCurrentExam} isDisabled={key === 'allowances'} />
       </Container>
-      <Tabs variant="tabs" mountOnEnter defaultActiveKey="attempts">
```
Comment (varshamenon4) [inline src/pages/ExamsPage/index.jsx:34]: Question: is there a reason that the Tabs component no longer needs the defaultActiveKey?

File: src/pages/ExamsPage/index.test.jsx [hunk 5]
```diff
@@ -20,6 +21,7 @@ jest.mock('./hooks', () => ({
 describe('ExamsPage', () => {
   beforeAll(() => {
     hooks.useExamAttemptsData.mockReturnValue(testUtils.defaultAttemptsData);
+    hooks.useAllowancesData.mockReturnValue({ allowancesList: [] });
```
Comment (varshamenon4) [inline src/pages/ExamsPage/index.test.jsx:24]: Question: I took a look at the epic and it looks like actually having allowances is not in the scope of this work, right? That is, for now, we only need to test this case and not the case where there are allowances? Thanks for clarifying.

File: src/pages/ExamsPage/__snapshots__/index.test.jsx.snap [hunk 6]
```diff
@@ -738,6 +747,15 @@ exports[`ExamsPage snapshots exams and attempts loaded 1`] = `
...
             Review Dashboard
           </a>
+          <a
+            aria-selected="false"
+            class="nav-item nav-link"
+            data-rb-event-key="allowances"
+            href="#"
+            role="tab"
+          >
+            Allowances
```
Comment (varshamenon4) [inline src/pages/ExamsPage/__snapshots__/index.test.jsx.snap:757]: Question: Is there a reason that "Allowances" isn't set as a message (for potential future internationalization, as with AllowanceList below) here? Is this just to be consistent with others (like Review Dashboard above)? Thanks for clarifying?

**Review by varshamenon4 — APPROVED** [review]
This looks awesome! Just a nit and a couple of questions.

File: src/pages/ExamsPage/messages.js (hunk 2 above)
Comment (alangsto) [reply src/pages/ExamsPage/messages.js:301]: Thank you!

File: src/pages/ExamsPage/__snapshots__/index.test.jsx.snap (hunk 6 above)
Comment (alangsto) [reply src/pages/ExamsPage/__snapshots__/index.test.jsx.snap:757]: Good question! I'm not entirely sure why the other tabs aren't translated, but yes, I did this to be consistent with them.

**Pull request merged**